CONTRIBUTING.MD

```diff
@@ -0,0 +1,97 @@
+
+```bash
+pre-commit install
+```
+
+To run the checks, some of which will automatically re-format your code to fit the standards, you can run
+```bash
+pre-commit run --all-files
+```
+It can also be run on a subset of files by omitting the `--all-files` option and pointing to specific files or folders.
+
+If you're using VS Code for development, pre-commit should setup git hooks that execute the pre-commit checks each time you check code into your branch through the integrated source-control as well. This will ensure that each of your commits conform to the desired format before they are run remotely and without needing to remember to run the checks before pushing to a remote. If this isn't done automatically, you can find instructions for setting up these hooks manually online.
+
+## Code Documentation
+
+For code documentation, we try to adhere to the Google docstring style (See [here](https://google.github.io/styleguide/pyguide.html), Section: Comments and Docstrings). The implementation of an extensive set of comments for the code in this repository is a work-in-progress. However, we are continuing to work towards a better commented state for the code. For development, as stated in the style guide, __any non-trivial or non-obvious methods added to the library should have a doc string__. For our library this applies only to code added to the main library in `fl4health`. Examples, research code, and tests need not incorporate the strict rules of documentation, though clarifying and helpful comments in that code is also __strongly encouraged__.
+
+__NOTE__: As a matter of convention choice, classes are documented through their `__init__` functions rather than at the "class" level.
+
+If you are using VS Code a very helpful integration is available to facilitate the creation of properly formatted docstrings called autoDocstring [VS Code Page](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) and [Documentation](https://github.com/NilsJPWerner/autoDocstring). This tool will automatically generate a docstring template when starting a docstring with triple quotation marks (`"""`). To get the correct format, the following settings should be prescribed in your VS Code settings JSON:
+
+```json
+{
+    "autoDocstring.customTemplatePath": "",
+    "autoDocstring.docstringFormat": "google",
+    "autoDocstring.generateDocstringOnEnter": true,
+    "autoDocstring.guessTypes": true,
+    "autoDocstring.includeExtendedSummary": false,
+    "autoDocstring.includeName": false,
+    "autoDocstring.logLevel": "Info",
+    "autoDocstring.quoteStyle": "\"\"\"",
+    "autoDocstring.startOnNewLine": true
+}
+```
+
+## Running Tests
```

Comment:
I think we can move the outstanding portion of the tests to contributing.md. It feels to me it would be best in one place and if someone is looking to run tests its probably because they are making a contribution. Maybe we can have a section in the main readme that points to contributing.md and gives a short sentence overview of what information is contained

Reply:
Yeah that makes sense. Will move now